src/Symfony/Component/Routing/Router.php

```diff
@@ -411,4 +414,15 @@ private function getConfigCacheFactory()
 
         return $this->configCacheFactory;
     }
+
+    private function checkDeprecatedOption($key)
+    {
+        switch ($key) {
+            case 'generator_base_class':
+            case 'generator_cache_class':
+            case 'matcher_base_class':
+            case 'matcher_cache_class':
+                @trigger_error(sprintf('Option "%s" given to router %s is deprecated since Symfony 4.2.', $key, static::class), E_USER_DEPRECATED);
```

Comment:
It is deprecated since 4.3 if this goes into `master`, right?

Reply:
good catch